probe-rs/src/vendor/mod.rs

```diff
@@ -119,9 +119,16 @@ fn try_detect_arm_chip(mut probe: Probe) -> Result<(Probe, Option<Target>), Erro
         // TODO: do not consume probe
         match probe.try_into_arm_interface() {
             Ok(interface) => {
-                let mut interface = interface
-                    .initialize(DefaultArmSequence::create(), dp_address)
-                    .map_err(|(_probe, err)| err)?;
+                let mut interface =
+                    match interface.initialize(DefaultArmSequence::create(), dp_address) {
+                        Ok(interface) => interface,
+                        Err((interface, error)) => {
+                            probe = interface.close();
+                            tracing::debug!("Error during ARM chip detection: {error}");
+                            // If we can't connect, assume this is not an ARM chip and not an error.
+                            return Ok((probe, None));
```

Comment:
I am not sure I agree with that logic. I would assume the Result to tell if any communication error occurred. The Option then says whether we could determine the target or not. In this case something went wrong with the communication (Result), not with the detection algorithm (Option)

Reply:
since it's device detection, we can't really tell most of the errors apart from the general case of talking chinese to something that doesn't understand

Reply:
Ah, I see the logic. Makes sense!